conntracksync/conntrack/conntrack.go

```diff
@@ -0,0 +1,166 @@
+package conntrack
+
+import (
+	//"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+)
+
+// CTEntry represents one entry from the conntrack table
+// Plucking only interested fields, ignoring others
+type CTEntry struct {
+	Protocol                string
+	OriginalSourceIP        string
+	OriginalDestinationIP   string
+	OriginalSourcePort      string
+	OriginalDestinationPort string
+	ReplySourceIP           string
+	ReplyDestinationIP      string
+	ReplySourcePort         string
+	ReplyDestinationPort    string
+}
+
+// Example:
+// tcp      6 431999 ESTABLISHED src=172.17.0.2 dst=172.22.101.201 sport=43009 dport=8080 src=172.22.101.201 dst=172.22.101.101 sport=8080 dport=43009 [ASSURED] mark=0 use=1
+//udp      17 173 src=10.49.61.42 dst=172.22.101.102 sport=4500 dport=4500 src=172.22.101.102 dst=172.22.101.101 sport=4500 dport=4500 [ASSURED] mark=0 use=1
+// tcp      6 65 TIME_WAIT src=172.22.101.1 dst=172.22.101.101 sport=59032 dport=9901 src=10.49.205.140 dst=172.22.101.1 sport=80 dport=59032 [ASSURED] mark=0 use=1
+// [ASSURED] or [UNREPLIED] can be present after the original IP/Port info
+// need to account for that
+const (
+	protocolIndex                = 0
+	originalSourceIPIndex        = 3
+	originalDestinationIPIndex   = 4
+	originalSourcePortIndex      = 5
+	originalDestinationPortIndex = 6
+	replySourceIPIndex           = 7
+	replyDestinationIPIndex      = 8
+	replySourcePortIndex         = 9
+	replyDestinationPortIndex    = 10
+)
+
+// ListDNAT lists only DNAT conntrack entries
+func ListDNAT() ([]CTEntry, error) {
+	return cmdCTListDNAT()
+}
+
+// CTEntryCreate Addetes the given entry from the conntrack table
+func CTEntryCreate(e CTEntry) error {
```

Comment:
I am using it in tests: https://github.com/rancher/plugin-manager/pull/62/commits/8ac914b3bb3f9f0403c305430c4f53f1737bb5c5#diff-337277601dc0c2f3bef9d8e80441356cR35